Source/Core/VideoCommon/VideoBackendBase.h

```diff
@@ -10,89 +10,59 @@
 
 #include "Common/CommonTypes.h"
 #include "Common/WindowSystemInfo.h"
-#include "VideoCommon/PerfQueryBase.h"
-
-namespace MMIO
-{
-class Mapping;
-}
-class PointerWrap;
+#include "VideoCommon/VideoBackendInfo.h"
+#include "VideoCommon/VideoCommon.h"
 
 class AbstractGfx;
 class BoundingBox;
 class Renderer;
 class TextureCacheBase;
 class VertexManagerBase;
-
-enum class FieldType
-{
-  Odd,
-  Even,
-};
-
-enum class EFBAccessType
-{
-  PeekZ,
-  PokeZ,
-  PeekColor,
-  PokeColor
-};
+class PerfQueryBase;
 
 class VideoBackendBase
 {
 public:
   virtual ~VideoBackendBase() {}
-  virtual bool Initialize(const WindowSystemInfo& wsi) = 0;
-  virtual void Shutdown() = 0;
+  virtual bool Initialize(const WindowSystemInfo& wsi) { return true; }
+  virtual void Shutdown() {}
 
   virtual std::string GetName() const = 0;
   virtual std::string GetDisplayName() const { return GetName(); }
   virtual void InitBackendInfo() = 0;
   virtual std::optional<std::string> GetWarningMessage() const { return {}; }
 
-  // Prepares a native window for rendering. This is called on the main thread, or the
-  // thread which owns the window.
-  virtual void PrepareWindow(WindowSystemInfo& wsi) {}
-
-  static std::string BadShaderFilename(const char* shader_stage, int counter);
+  // The various subclassed components that backends must supply
+  virtual std::unique_ptr<AbstractGfx> CreateGfx() = 0;
```

Comment:
Should all these `Create` functions be const?

Reply:
I think all except CreateGfx can be made const... but not in this PR.

CreateGfx kind of needs to modify the backend state.